shopgate/pwa
# Implemented the EmbeddedMedia collection to improve handling of playing media on navigation

# Description
Within single page applications, pages are never really left on navigation. So special handling is necessary for playing media to avoid that it keeps playing within the background.

The html helpers of the `common` library already included logic to handle YouTube videos, but the system wasn't extensible.

This pull request introduces the `EmbeddedMedia` collection. It combines some approaches from the (now deprecated) YouTube helper with the collection system which will be introduced with `PWA 6.0`. The collection allows to register `media providers` which can handle specific embedded media. For now `YouTube` and `Vimeo` is integrated.

Wherever embedded media occurs - usually within components that display HTML blocks from API responses - the respective DOM containers can be registered within the collection. The necessary logic for e.g. route changes is handled internally within the collection.

## Type of change

- [  ] Bug Fix :bug: (non-breaking change which fixes an issue)
- [ x ] New Feature :rocket: (non-breaking change which adds functionality)
- [  ] Breaking Change :boom: (fix or feature that would cause existing functionality to not work as expected)
- [  ] Polish :nail_care: (Just some cleanups)
- [  ] Docs :memo: (Changes in the documentations)
- [  ] Internal :house: Only relates to internal processes.

## How to test it

For testing you need to setup one or more HTML widgets with `YouTube` and `Vimeo` videos. Simply copy & paste some snippets from the `Share` section of the different video providers.
After opening the app, navigate to the particular pages where the widgets are configured and start playing. When you leave the page in any direction, the videos should stop playing.
